system/Commands/Generators/CreateModel.php

```diff
@@ -0,0 +1,188 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @package    CodeIgniter
+ * @author     CodeIgniter Dev Team
+ * @copyright  2019-2020 CodeIgniter Foundation
+ * @license    https://opensource.org/licenses/MIT	MIT License
+ * @link       https://codeigniter.com
+ * @since      Version 4.0.0
+ * @filesource
+ */
+
+namespace CodeIgniter\Commands\Generators;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use Config\Services;
+
+/**
+ * Creates a new model file.
+ *
+ * @package CodeIgniter\Commands
+ */
+class CreateModel extends BaseCommand
+{
+	/**
+	 * The group the command is lumped under when listing commands.
+	 *
+	 * @var String
+	 */
+	protected $group = 'Generators';
+
+	/**
+	 * The Command's name
+	 *
+	 * @var String
+	 */
+	protected $name = 'make:model';
+
+	/**
+	 * The Command's short description
+	 *
+	 * @var String
+	 */
+	protected $description = 'Creates a new model file.';
+
+	/**
+	 * The Command's usage
+	 *
+	 * @var String
+	 */
+	protected $usage = 'make:model [name] [options]';
+
+	/**
+	 * The Command's arguments
+	 *
+	 * @var Array
+	 */
+	protected $arguments = [
+		'name' => 'The model file name',
+	];
+
+	/**
+	 * The Command's options
+	 *
+	 * @var Array
+	 */
+	protected $options = [
+		'-n' => 'Set model namespace',
+		'-t' => 'Set the table name',
+		'-e' => 'Set the entity name',
+	];
+
+	/**
+	 * Creates a new model file.
+	 *
+	 * @param Array $params
+	 */
+	public function run(Array $params): void
+	{
+		helper('inflector');
+
+		$name = array_shift($params);
+
+		if (empty($name))
+		{
+			$name = CLI::prompt(lang('Model.nameFile'), null, 'required');
+		}
+
+		// Get options
+		$ns       = $params['-n'] ?? CLI::getOption('n');
+		$table    = $params['-t'] ?? CLI::getOption('t');
+		$entity   = $params['-e'] ?? CLI::getOption('e');
+
+		$homepath = APPPATH;
+
+		if (! empty($ns))
+		{
+			// Get all namespaces
+			$namespaces = Services::autoloader()->getNamespace();
+
+			foreach ($namespaces as $namespace => $path)
+			{
+				if ($namespace === $ns)
+				{
+					$homepath = realpath(reset($path)) . DIRECTORY_SEPARATOR;
+					break;
+				}
+			}
+		}
+		else
+		{
+			$ns = defined('APP_NAMESPACE') ? APP_NAMESPACE : 'App';
+		}
+
+		// Full path
+		$path = $homepath . 'Models/' . $name . '.php';
+
+		// Class name should be pascal case now (camel case with upper first letter)
+		$name = pascalize($name);
+
+		$template = "<?php namespace $ns\Models;\n\n" .
+
+'use CodeIgniter\Model;
+
+class {name} extends Model
+{
+	protected $table              = "{table}";
+	protected $primaryKey         = "id";
+
+	protected $returnType         = "{entity}";
+	protected $useSoftDeletes     = true;
+
+	protected $allowedFields      = [];
+
+	protected $useTimestamps      = false;
+	protected $createdField       = "created_at";
+	protected $updatedField       = "updated_at";
+	protected $deletedField       = "deleted_at";
```

Comment:
These are the default fields and are probably unlikely to change for most models, or be changed in a `BaseModel` file. I recommend leaving them out, unless you have a compelling reason for including them.